pkgs/os-specific/linux/swapspace/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchgit, autoconf, automake, utillinux, btrfs-progs }:
+
+stdenv.mkDerivation rec {
+  version = "1.16.1";
+  name = "swapspace-${version}";
+
+  src = fetchgit {
+    url = "https://github.com/Tookmund/Swapspace.git";
+    rev = "refs/tags/v${version}";
+    sha256 = "1hbxaq05zdfhw5c8lypnl1qvh7n9ng0nawlizs9g6l77z8sn0ji8";
+    fetchSubmodules = false;
+  };
+
+  # TODO patch runcommands to include path to mkswap, swapon/off, btrfs (optional)
```

Comment:
That TODO is leftover from the `patchPhase` below?

sed tends to silently break and not apply anymore.

To make this more robust, we usually provide a patch that replaces sources with placeholders, and then do a `substituteAll` on the patch file with the actual occurences - see the bazel derivation for an example.

Reply:
Or use `substituteInPlace` which will complain if a substitution cannot be made.

Reply:
`substituteInPlace` doesn't fail, just complains to stdout and silently regresses. `substituteAll` on a patch is the cleaner way to do this.

Reply:
The btrfs executable is no longer used and the other invocations are very straightforward and likely to keep working across versions, so I opted to keep this approach for easier maintenance.